src/bootstrap/tarball.rs

```diff
@@ -297,8 +297,9 @@ impl<'a> Tarball<'a> {
     fn run(self, build_cli: impl FnOnce(&Tarball<'a>, &mut Command)) -> GeneratedTarball {
         t!(std::fs::create_dir_all(&self.overlay_dir));
         self.builder.create(&self.overlay_dir.join("version"), &self.overlay.version(self.builder));
-        if let Some(sha) = self.builder.rust_sha() {
-            self.builder.create(&self.overlay_dir.join("git-commit-hash"), &sha);
+        if let Some(info) = self.builder.rust_info.info() {
+            let commit_info = format!("{}\n{}\n{}\n", info.sha, info.short_sha, info.commit_date);
```

Comment:
This format is duplicated in two places for writing, and one place for reading. Can we factor the read/write into one location (e.g., methods on a struct or similar) so that it's much easier to remember to update? This is particularly annoying since the code as written silently ignores the produced file, so we're unlikely to get bug reports (just unfortunate behavior).

Reply:
> Can we factor the read/write into one location (e.g., methods on a struct or similar) so that it's much easier to remember to update?

I think this would fit well on the `Build` struct in `src/boostrap/lib.rs`, so I'll add methods for reading from and writing to the file.

> This is particularly annoying since the code as written silently ignores the produced file, so we're unlikely to get bug reports (just unfortunate behavior).

Do you think it'd be better to simply error if the formatting is wrong (assuming you're referring to [this code](https://github.com/dawnofmidnight/rust/blob/0561ae355cef0fa2a32b6bb09730ca03879c92f6/src/bootstrap/channel.rs#L49)).

Reply:
Yes, I think so. In general, failing loudly is my preference or at least logging *something* (perhaps only in verbose mode, depending on how common it is to fail).

Reply:
As for the former issue, I'm not sure of a good way to make a method on `Build` and then let `channel::GitInfo::new` use it if necessary. Any suggestion on that, or for a better place?

Reply:
@dawnofmidnight make it `pub(crate)`